x/incentives/keeper/gauge.go

```diff
@@ -298,6 +294,35 @@ func (k Keeper) CreateGroup(ctx sdk.Context, coins sdk.Coins, numEpochPaidOver u
 	return groupGaugeID, nil
 }
 
+// chargeGroupCreationFeeIfNotWhitelisted charges fee as defined in the params if the sender is not whitelisted.
+// Does not charge fee if sender is the incentives module account or if sender is whitelisted.
+// Returns true if charged fee, false otherwise.
+// Returns error if:
+// - One if the addresses in params is invalid
+// - fails to send coins from sender to the community pool
+func (k Keeper) chargeGroupCreationFeeIfNotWhitelisted(ctx sdk.Context, sender sdk.AccAddress) (chargedFee bool, err error) {
+	params := k.GetParams(ctx)
+	incentivesModuleAddress := k.ak.GetModuleAddress(types.ModuleName)
+	for _, unrestrictedAddressStr := range params.UnrestrictedCreatorWhitelist {
+		unrestrictedAddress, err := sdk.AccAddressFromBech32(unrestrictedAddressStr)
+		if err != nil {
+			return false, err
+		}
+
+		// sender is either unrestrictedAddress or the module account
+		if unrestrictedAddress.Equals(sender) || sender.Equals(incentivesModuleAddress) {
+			return false, nil
+		}
+	}
```

Comment:
wdyt about adding the module address to this whitelist param instead of retrieving it on every call? Can ignore, would be minor perf.

Reply:
I prefer the explicitness of getting a module account for readability. If it's in params, it's not obvious and might make it a hard-to-debug detail in the future